src/services/transaction_watcher_service.ts

```diff
@@ -0,0 +1,116 @@
+                    logger.trace({
+                        message: `a transaction with a ${
+                            txEntity.status
+                        } status is already on the blockchain, updating status to confirmed`,
+                        hash: txInBlockchain.hash,
+                    });
+                    txEntity.status = TransactionStates.Confirmed;
+                    txEntity.blockNumber = txInBlockchain.blockNumber;
+                    await this._transactionRepository.save(txEntity);
+                    await this._abortTransactionsWithTheSameNonceAsync(txEntity);
+                    return txEntity;
+                    // Checks if the txn is in the mempool but still has it's status set to Unsubmitted or Submitted
+                } else if (!isExpired && txEntity.status !== TransactionStates.Mempool) {
+                    logger.trace({
+                        message: `a transaction with a ${
+                            txEntity.status
+                        } status is pending, updating status to mempool`,
+                        hash: txInBlockchain.hash,
+                    });
+                    txEntity.status = TransactionStates.Mempool;
+                    return this._transactionRepository.save(txEntity);
+                } else if (isExpired) {
+                    // NOTE(oskar): we currently cancel all transactions that are in the
+                    // "stuck" state. A better solution might be to unstick
+                    // transactions one by one and observing if they unstick the
+                    // subsequent transactions.
+                    txEntity.status = TransactionStates.Stuck;
+                    return this._transactionRepository.save(txEntity);
+                }
+            }
+        } catch (err) {
+            if (err instanceof TypeError) {
+                // TODO(oskar) `getTransactionByHashAsync` throws a TypeError
+                // when tx returned from `eth_getTransactionByHash` is null.
+                if (isExpired) {
+                    txEntity.status = TransactionStates.Dropped;
+                    return this._transactionRepository.save(txEntity);
+                }
+            } else {
+                // if the error is not from a typeerror, we rethrow
+                throw err;
+            }
+        }
+        return txEntity;
+    }
+    // Sets the transaction status to 'aborted' for transactions with the same nonce as the passed in txEntity
+    private async _abortTransactionsWithTheSameNonceAsync(txEntity: TransactionEntity): Promise<TransactionEntity[]> {
+        const transactionsToAbort = await this._transactionRepository.find({
+            where: {
+                nonce: txEntity.nonce,
+                hash: Not(txEntity.hash),
+                // if the transaction has already been marked as dropped, we can skip it
+                status: Not(TransactionStates.Dropped),
```

Comment:
Let's overwrite `Dropped` to `Aborted` because we are 100% sure these txns won't ever get mined. That to me, is the fundamental difference between `Dropped` and `Aborted`.